### PR TITLE
[Merged by Bors] - feat: port Order.Bounds.Basic

### PR DESCRIPTION
**Pull request by mcdoll**
aba57d4d3dae35460225919dcd82fe91355162f9

### CODE REVIEW

**Review by ChrisHughes24 — APPROVED** [review]

File: Mathlib/Order/Bounds/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,1645 @@
...
+
+theorem IsLeast.lowerBounds_eq (h : IsLeast s a) : lowerBounds s = Iic a :=
+  h.isGLB.lowerBounds_eq
+#align is_least.lower_bounds_eq IsLeast.lowerBounds_eq
+
+theorem IsGreatest.upperBounds_eq (h : IsGreatest s a) : upperBounds s = Ici a :=
+  h.isLUB.upperBounds_eq
+#align is_greatest.upper_bounds_eq IsGreatest.upperBounds_eq
+
+theorem is_lub_le_iff (h : IsLUB s a) : a ≤ b ↔ b ∈ upperBounds s := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Bounds/Basic.lean:301]: I don't think this name is right, but I'm not sure if it should say `isLub_le_iff` or `isLUB_le_iff`. Seems like you did `isLUB` in some places below.

File: Mathlib/Order/Bounds/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,1645 @@
...
+
+
+namespace MonotoneOn
+
+variable [Preorder α] [Preorder β] {f : α → β} {s t : Set α} (Hf : MonotoneOn f t) {a : α}
+  (Hst : s ⊆ t)
+
+-- Porting note: include is deprecated
+--include Hf
+
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Bounds/Basic.lean:1138]: I think this is fine to remove
```suggestion
```

File: Mathlib/Order/Bounds/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,1645 @@
...
+end MonotoneOn
+
+namespace AntitoneOn
+
+variable [Preorder α] [Preorder β] {f : α → β} {s t : Set α} (Hf : AntitoneOn f t) {a : α}
+  (Hst : s ⊆ t)
+
+-- Porting note: include is deprecated
+--include Hf
+
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Bounds/Basic.lean:1203]:
```suggestion
```

(some more below)

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**